conn.go

```diff
@@ -94,7 +94,7 @@ func (c wrappedConn) Query(query string, args []driver.Value) (driver.Rows, erro
 		if err != nil {
 			return nil, err
 		}
-		return wrappedRows{intr: c.intr, parent: rows}, nil
+		return wrapRows(nil, c.intr, rows), nil //nolint
```

Comment:
If I understand it right, the nil context that is passed here is then forwarded to the intercepted rows methods?
If this is the case this will cause nil ptr dereference panics for users.
Users of the lib can assume that ctx is never nil because the stdlib context document says it must never be nil. (https://pkg.go.dev/context)
```suggestion
		return wrapRows(context.Background(), c.intr, rows), nil //nolint
```